### ISSUE TITLE
Notarize sqlectron for macOS

### ISSUE THREAD

**Issue by connorads** [sqlectron/sqlectron-gui#527]
`sqlectron` is currently not [notarized](https://developer.apple.com/documentation/xcode/notarizing_macos_software_before_distribution), which means that it is difficult to run on modern macOS versions [like Catalina due to enhanced security features](https://support.apple.com/en-euro/HT202491).

![image](https://user-images.githubusercontent.com/10026538/95997317-928c8f80-0e2b-11eb-8f75-eac2f4d83799.png)

- [Notarizing macOS Software Before Distribution](https://developer.apple.com/documentation/xcode/notarizing_macos_software_before_distribution)
- [Apple Developer Program Membership Fee Waivers](https://developer.apple.com/support/membership-fee-waiver/)

**Comment by MasterOdin** [comment]
Yup, that is a good point. I have been meaning to look into this for both windows and macOS to enable an easier time for installation. I have added this to the 1.33.0 backlog to accomplish.

**Comment by maxcnunes** [comment]
When I initially implemented the release for Mac I couldn't find a way of having a free Apple developer license. With that last link, it seems to be possible, I will see if I can get that to work 🤞

**Comment by MasterOdin** [comment]
When I looked into this a month ago, looking at the site, by "non-profit organization", I almost certainly believe that they mean an organization that is registered as a 501(c) in the US (or similarly in other supported countries), and not just a GH organization that distributes free software.

Unfortunately, I think that means we'd have to pony up the $100 a year for this, which I'm happy to do, especially as I wanted to be able to notarize other software I was developing for the mac platform during this year.

**Comment by maxcnunes** [comment]
Ah, it was too good to be real. Since we don't have any income from this app, I don't think we should pay from our pockets. I think we could open sponsorship so users can help to maintain the project.

**Comment by dsecareanu** [comment]
I've tried to install the app today on MacOS Big Sur to test it out and I ran into the same issue. Yes, you should open sponsorship on github, I am sure there will be people who will support your work.

**Comment by manngo** [comment]
> Yes, you should open sponsorship on github, I am sure there will be people who will support your work.

Yes, I’d be in that.

**Comment by DiyazY** [comment]
Hi. I was able to overcome an issue with permissions. However, It just doesn't work on Bug Sur MacOS.
![image](https://user-images.githubusercontent.com/16912141/104838788-d331f400-58c5-11eb-8570-f5fb9769df0e.png)
Probably, It should be a separate issue 😊

Here are the details:
Process:               sqlectron [32071]
Path:                  /Applications/sqlectron.app/Contents/MacOS/sqlectron
Identifier:            org.sqlectron.gui
Version:               1.32.1 (1.32.1)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           sqlectron [32071]
User ID:               501

Date/Time:             2021-01-17 13:04:35.190 +0200
OS Version:            macOS 11.1 (20C69)
Report Version:        12
Bridge OS Version:     5.1 (18P3030)
Anonymous UUID:        AA0B245C-7221-DD3A-44F7-DC497065E9CE

Sleep/Wake UUID:       E42C01DC-978D-4C39-882F-52F3D38490B3

Time Awake Since Boot: 160000 seconds
Time Since Wake:       2000 seconds

System Integrity Protection: enabled

Crashed Thread:        0

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    DYLD, [0x1] Library missing

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  dyld: Using shared cache: 467A83CB-BA86-3F07-B652-B9256C74080A
Library not loaded: @rpath/Electron Framework.framework/Electron Framework
  Referenced from: /Applications/sqlectron.app/Contents/MacOS/sqlectron
  Reason: no suitable image found.  Did find:
	/Applications/sqlectron.app/Contents/MacOS/../Frameworks/Electron Framework.framework/Electron Framework: file too short

Binary Images:
       0x107afa000 -        0x107b22fff +org.sqlectron.gui (1.32.1 - 1.32.1) <596191EB-2C06-339C-917C-D986342C029E> /Applications/sqlectron.app/Contents/MacOS/sqlectron
       0x117281000 -        0x11731cfff  dyld (832.7.1) <DEA51514-B4E8-3368-979B-89D0F8397ABC> /usr/lib/dyld
    0x7fff3eaeb000 -     0x7fff3ec42fff  com.apple.MediaPlayer (1.0 - 1.0) <98F74E6D-4344-3FC0-9148-00E29BBDB37D> /System/Library/Frameworks/MediaPlayer.framework/Versions/A/MediaPlayer

Model: MacBookPro16,1, BootROM 1554.60.15.0.0 (iBridge: 18.16.13030.0.0,0), 8 processors, 8-Core Intel Core i9, 2,4 GHz, 64 GB, SMC 
Graphics: kHW_IntelUHDGraphics630Item, Intel UHD Graphics 630, spdisplays_builtin
Graphics: kHW_AMDRadeonPro5300MItem, AMD Radeon Pro 5300M, spdisplays_pcie_device, 4 GB
Memory Module: BANK 0/ChannelA-DIMM0, 32 GB, DDR4, 2667 MHz, Micron, -
Memory Module: BANK 2/ChannelB-DIMM0, 32 GB, DDR4, 2667 MHz, Micron, -
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x7BF), wl0: Sep 11 2020 17:01:15 version 9.30.440.2.32.5.61 FWID 01-1d69e4b4
Bluetooth: Version 8.0.2f9, 3 services, 27 devices, 1 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
USB Device: USB 3.1 Bus
USB Device: Apple T2 Bus
USB Device: Composite Device
USB Device: Touch Bar Backlight
USB Device: Touch Bar Display
USB Device: Apple Internal Keyboard / Trackpad
USB Device: Headset
USB Device: Ambient Light Sensor
USB Device: FaceTime HD Camera (Built-in)
USB Device: Apple T2 Controller
Thunderbolt Bus: MacBook Pro, Apple Inc., 63.1
Thunderbolt Bus: MacBook Pro, Apple Inc., 63.1

**Comment by dsecareanu** [comment]
Actually I've found out somewhere else that you can open non-notarized apps. You just need to go and (hold ctrl and) right click on the app in the Applications folder and will ask you if you want to open it or not, but you do have an Open button which works (macbook pro 16, big sur 11.1).

Found an article here: https://eclecticlight.co/2020/11/19/getting-unnotarized-apps-out-of-quarantine/ (not my initial source, as that one was a github thread somewhere).
